msprime/trees.py

```diff
@@ -289,17 +289,13 @@ def __init__(
         self._tree = tree
         self._x_coords = {}
         self._y_coords = {}
-        try:
-            self._branch_colours = {node: col for node, col in branch_colours.keys()}
-        except AttributeError:
-            self._branch_colours = {node: col for node, col in enumerate(branch_colours)}
-        
+        self._branch_colours = branch_colours or {}
```

Comment:
Oh, yuck. Didn't know that about pep8 verbosity. Will change against my normal programming preference for short-circuiting.